compiler-plugin/src/main/java/io/ballerina/stdlib/websocket/plugin/Utils.java

```diff
@@ -94,7 +94,8 @@ public static void validateOnBinaryMessageFunction(FunctionTypeSymbol functionTy
             for (ParameterSymbol inputParam : inputParams) {
                 String moduleId = getModuleId(inputParam);
                 String paramSignature = inputParam.typeDescriptor().signature();
-                if (!paramSignature.equals(BYTE_ARRAY) && !paramSignature.equals(moduleId + COLON + CALLER)) {
+                if (!paramSignature.equals(BYTE_ARRAY) && !paramSignature.equals(moduleId + COLON + CALLER) &&
+                        !(inputParam.typeDescriptor().typeKind() == TypeDescKind.INTERSECTION)) {
```

Comment:
Does this mean going forward we only support `readonly & byte[] data` ?

Reply:
No, this only says it doesn't give any compiler plugin error when we have `readonly & byte[] data`. It supports both `byte[]` and `readonly & byte[] data`.

Reply:
Nice!